lib/Config/Config.cpp

```diff
@@ -299,6 +305,14 @@ void Config::parseServerMessage()
         m_nodeMode = static_cast<NodeModes>(m_packetBuffer.at(value(Offsets::PAYLOAD_START_OFFSET)));
         m_nodeModeHash = m_lastHash;
         break;
+    case NAD:
+        if (1 != message_size)
+        {
+            return;
+        }
+        m_nad = static_cast<uint8_t>(m_packetBuffer.at(value(Offsets::PAYLOAD_START_OFFSET)));
```

Comment:
You are reading from the packetbuffer at the same position as the NodeMode is stored. 
As you can see in line 299, you are reading from Offset::PAYLOAD_START_OFFSET. 
I think you need to change to the correct offset for nad.